### PR TITLE
feat(testutils): add comprehensive generic testing framework

### PR DESCRIPTION
**Pull request by amery**
## Summary

Introduces `darvaza.org/x/testutils`, a comprehensive generic testing framework providing systematic TestCase types for Go method, factory, and function testing patterns. This package enables type-safe, composable test utilities that reduce boilerplate whilst maintaining excellent test clarity.

- **60 generic TestCase types** across 3 testing patterns with 0-5 argument support
- **Complete meta-testing infrastructure** ensuring framework reliability
- **Memory-optimised implementations** with developer-friendly factory functions
- **Extensive documentation** with practical usage examples
- **Seamless integration** with existing `core.TestCase` interface

### Three-Pattern Design

1. **Method Testing**: Functions taking `*T` as parameter (18 types)
2. **Factory Testing**: Functions returning `*T` constructors (18 types)
3. **Function Testing**: Functions returning comparable values (18 types)

### Systematic Code Generation

- **Code generation script** (`generic_gen.sh`) creates all 60 TestCase implementations
- **Argument variants**: Base, OneArg, TwoArgs, ThreeArgs, FourArgs, FiveArgs
- **Memory alignment**: Field-optimised structs with logical parameter ordering

### Meta-Testing Infrastructure

- **Success/failure validation** for all 60 TestCase implementations
- **Comprehensive test coverage** (5,000+ lines of test code)
- **Isolated test execution** using `core.MockT` for sandboxed validation

### Package Structure

- **New Tier 1 package**: Independent module with only `darvaza.org/core` dependency
- **Root documentation updates**: AGENTS.md, README.md, RELEASE.md integration
- **Build system integration**: cspell dictionary and release procedures

### Usage Benefits

- **Composable testing**: Create reusable test functions covering multiple methods
- **Type safety**: Full generic support with compile-time validation
- **Reduced boilerplate**: Factory functions hide complexity behind clean interfaces
- **Systematic coverage**: Consistent patterns across all argument counts

## Test Plan

- [x] All 60 TestCase types generate and compile correctly
- [x] Complete meta-testing suite passes (`make test-testutils`)
- [x] Code quality compliance (`make tidy-testutils`)
- [x] Documentation completeness and accuracy
- [x] Integration with mono-repo build system
- [x] Memory alignment and performance optimisation
- [x] Example usage demonstrations work correctly

___

### **PR Type**
Enhancement, Tests, Documentation

___

### **Description**
• Introduces `darvaza.org/x/testutils`, a comprehensive generic testing framework with 60 TestCase types across method, factory, and function testing patterns
• Implements systematic code generation for all TestCase variants supporting 0-5 arguments with memory-optimized struct layouts
• Provides complete meta-testing infrastructure with 5,000+ lines of test code validating all framework components
• Adds extensive documentation including README, AGENTS.md, and comprehensive usage examples
• Integrates with existing `core.TestCase` interface while maintaining type safety and composability
• Updates project documentation and build system to include the new Tier 1 package
• Includes utility functions for test execution, data generation, and framework validation

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["Code Generation Script"] --> B["60 Generic TestCase Types"]
  B --> C["Method Testing (18 types)"]
  B --> D["Factory Testing (18 types)"]
  B --> E["Function Testing (18 types)"]
  F["Meta-Testing Suite"] --> G["Success/Failure Validation"]
  G --> H["Framework Reliability"]
  I["Documentation"] --> J["Usage Examples"]
  I --> K["API Reference"]
  L["Build Integration"] --> M["Project Updates"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><details><summary>8 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>generic_getter_test.go</strong><dd><code>Add comprehensive getter pattern TestCase validation suite</code></dd></summary>
<hr>

testutils/generic_getter_test.go

• Comprehensive test suite for all 18 getter-pattern TestCase types <br>with 0-5 argument variants<br> • Tests success and failure scenarios for <br>basic getters, OK-pattern getters, and error-pattern getters<br> • <br>Includes edge cases like empty values, unicode characters, boundary <br>conditions, and error type validation<br> • Uses <code>TestStruct</code> methods to <br>validate getter behavior across different return patterns

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-24aa0b4aaeeb214a9364b1529947d2999fbc539926a0637c515e669e7456f5b0">+1327/-0</a></td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>run_test.go</strong><dd><code>Add comprehensive test runner function validation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

testutils/run_test.go

• Tests for <code>Run()</code> function covering all three interface branches <br>(testing.T, core.T, fallback)<br> • Comprehensive validation of <br><code>RunSuccessCases()</code> and <code>RunFailureCases()</code> with various pass/fail <br>scenarios<br> • Mock implementations for different T interface types to <br>test isolation and proper result reporting<br> • Integration tests with <br>real generated TestCase types to verify end-to-end functionality

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-64b009c3b0bc3fc8ea87b8662f9182657513a33f0f6fe45d63c11ee0b903f897">+619/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>utils_test.go</strong><dd><code>Add utility functions and data structures test coverage</code>&nbsp; &nbsp; </dd></summary>
<hr>

testutils/utils_test.go

• Tests for <code>ArrayTestCaseData</code> factory and all its methods (indexes, <br>names, Make)<br> • Validation of <code>DummyTestCase</code> behavior for both passing <br>and failing scenarios<br> • Tests for utility functions like <code>RunTest</code>, <br><code>UniqueSliceOrderedFn</code>, and convenience helpers<br> • Edge case testing <br>including empty arrays, duplicate indexes, and out-of-range values

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-ecad2c77ae2da9cc51a8ef993301d7df2fb6b7d3501bd5735b0da525c0f7470c">+400/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>generic_factory_test.go</strong><dd><code>Factory TestCase comprehensive test suite implementation</code>&nbsp; </dd></summary>
<hr>

testutils/generic_factory_test.go

• Comprehensive test suite for 18 factory TestCase types with 0-5 <br>argument variants<br> • Tests both success and failure scenarios for <br>factory functions returning *T, (*T, bool), and (*T, error)<br> • Includes <br>validation testing with <code>validateTestStruct</code> and nil expectation <br>handling<br> • Meta-testing infrastructure using <code>testutils.RunSuccessCases</code> <br>and <code>testutils.RunFailureCases</code>

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-b0308fa085256753b02c8fa642bf33d04ef1f6a30e720fd835237c13e6f9e51b">+958/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>generic_function_test.go</strong><dd><code>Function TestCase comprehensive test suite implementation</code></dd></summary>
<hr>

testutils/generic_function_test.go

• Complete test coverage for 18 function TestCase types supporting 0-5 <br>arguments<br> • Tests functions returning V, (V, bool), and (V, error) <br>with various data types<br> • Includes mathematical operations, string <br>processing, and error handling scenarios<br> • Demonstrates function <br>testing patterns with predictable outputs and edge cases

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-de9480f7726d8aa4a111cf875b9177fecabfcfec664e91af15512fa137853647">+949/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>utils.go</strong><dd><code>Testing utilities and meta-testing infrastructure</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

testutils/utils.go

• <code>DummyTestCase</code> implementation for meta-testing with configurable <br>pass/fail behavior<br> • <code>ArrayTestCaseData</code> for generating test case arrays <br>with controlled failure patterns<br> • Utility functions for test <br>execution, index sanitization, and slice operations<br> • Helper functions <br>for creating passing/failing test cases with formatted names

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-b96eabf908a6defb59888cfa27d524accdbf715237b70e3784c8a5a22680e714">+216/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>utils_private_test.go</strong><dd><code>Private utility function unit test</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

testutils/utils_private_test.go

• Unit test for <code>matchStringDummy</code> helper function used by <br><code>testing.RunTests</code><br> • Validates the dummy match function always returns <br>true with no errors

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-f3b7cd1fe8839e7e1dbc56b2c86a2fa64ca8d550e971a9a88540a3a75a83e81e">+17/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>generic_common_test.go</strong><dd><code>Add comprehensive test fixtures and factory functions for generic </code><br><code>testing</code></dd></summary>
<hr>

testutils/generic_common_test.go

• Defines comprehensive test data structures and methods for testing <br>the generic testing framework<br> • Implements <code>TestStruct</code> with various <br>method signatures (getters, validators, combiners) supporting 0-5 <br>arguments<br> • Provides factory functions for creating <code>TestStruct</code> <br>instances with different patterns (basic, OK, Error variants)<br> • <br>Includes predefined error constants for systematic error testing <br>scenarios

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-4d532de28821cab905485a2ffc70c7aa258f579335c5dacef42243e3dbb8c25c">+356/-0</a>&nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Enhancement</strong></td><td><details><summary>3 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>run.go</strong><dd><code>Implement test runner functions with interface adaptation</code></dd></summary>
<hr>

testutils/run.go

• Implements <code>Run()</code> function with type assertion logic for different T <br>interface types<br> • Provides <code>RunSuccessCases()</code> and <code>RunFailureCases()</code> for <br>meta-testing TestCase implementations<br> • Uses sandboxed execution via <br><code>RunTest</code> to isolate test case validation<br> • Includes helper function <br><code>doRunTests()</code> for common test execution patterns

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-fbd3d39a3891cfa780b14c273ccdb3d2b7cd040510bb648e051282bacf98f683">+95/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>generic.go</strong><dd><code>Add generic type validation function interface</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

testutils/generic.go

• Defines <code>TypeTestFunc</code> generic type for factory-created object <br>validation functions<br> • Provides type-safe validation interface for <br>testing constructed objects

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-6cf5e0f15a94b18d2d4ddd6d6dabda71a82e1b56a0fe0f7486c8ab265a364371">+6/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>generic_gen.sh</strong><dd><code>Add code generation script for systematic TestCase type creation</code></dd></summary>
<hr>

testutils/generic_gen.sh

• Implements comprehensive code generation script for creating 60 <br>generic TestCase types<br> • Generates systematic variations for 0-5 <br>argument support across method, factory, and function patterns<br> • <br>Provides helper functions for struct generation, field alignment, and <br>constructor creation<br> • Creates type definitions, verification <br>statements, and complete TestCase implementations

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-c517a92b80f2f98063d5793872e4f0dfcb0ecab2572bb0c712291e89d49829e7">+1112/-0</a></td>

</tr>
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>7 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>example_test.go</strong><dd><code>Comprehensive testing framework usage examples and documentation</code></dd></summary>
<hr>

testutils/example_test.go

• Extensive example demonstrations of all three testing patterns <br>(method, factory, function)<br> • Calculator domain model with validation <br>functions and custom factory helpers<br> • Real-world usage patterns <br>showing error handling, type validation, and custom test factories<br> • <br>Educational examples for comprehensive testing framework usage

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-2108189e0598534d99793f3fd7c7f51c03862786ec752ff9e9d0da47b4bfa186">+374/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>doc.go</strong><dd><code>Add comprehensive package documentation with usage examples and </code><br><code>patterns</code></dd></summary>
<hr>

testutils/doc.go

• Provides comprehensive package documentation explaining three <br>testing patterns (method, factory, function)<br> • Documents function type <br>patterns and their generic constraints for different argument counts<br> • <br>Includes extensive usage examples demonstrating TestCase creation and <br>execution with <code>core.RunTestCases</code><br> • Explains integration with the <br><code>core.TestCase</code> interface and testing framework patterns

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-332b86d465498a310091744b2b4c0c3f86a4c1617e4e67fea1b7707f6e7e5a74">+146/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>README.md</strong><dd><code>Add comprehensive README with API documentation and usage examples</code></dd></summary>
<hr>

testutils/README.md

• Provides detailed package overview explaining generic testing <br>framework architecture<br> • Documents all available TestCase types, <br>factory functions, and testing utilities<br> • Includes comprehensive <br>usage examples for method, factory, and function testing patterns<br> • <br>Explains integration with <code>core.TestCase</code> interface and development <br>workflow

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-94ed328faa121d1efabf97a401989e3165ee3761fbf5518f11b9f5c1973f315d">+481/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>AGENTS.md</strong><dd><code>Add agent documentation for development guidelines and architecture</code></dd></summary>
<hr>

testutils/AGENTS.md

• Provides detailed agent documentation explaining package <br>architecture and design patterns<br> • Documents core components, testing <br>patterns, and code generation approach<br> • Includes development <br>guidelines, performance considerations, and integration notes<br> • <br>Explains field alignment principles and factory function requirements

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-9ef84dbafeddc4fc5e27a7cc822e13e57b7705f0361a5608573bd69f33247255">+448/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>RELEASE.md</strong><dd><code>Update release documentation to include testutils package</code></dd></summary>
<hr>

RELEASE.md

• Adds <code>testutils</code> package to Tier 1 independent packages in release <br>documentation<br> • Updates dependency diagrams and release procedures to <br>include the new package<br> • Modifies release commands and version <br>checking to incorporate <code>testutils</code>

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-2b1b69303b927a484e02c7fad9fc87d0d3ff0dc22ae1da0ecd0dc935d922a23c">+25/-22</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>README.md</strong><dd><code>Add testutils package to main project documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

README.md

• Adds <code>testutils</code> section describing generic test case types and <br>testing utilities<br> • Updates dependency diagram to include <code>testutils</code> as <br>Tier 1 package<br> • Includes <code>testutils</code> in package listings and reference <br>links

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5">+12/-1</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>AGENTS.md</strong><dd><code>Add testutils reference to main agent documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

AGENTS.md

• Adds reference to <code>testutils/AGENTS.md</code> in the package-specific <br>documentation list

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-a54ff182c7e8acf56acfd6e4b9c3ff41e2c41a31c9b211b2deb9df75d9a478f9">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Miscellaneous</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>LICENCE.txt</strong><dd><code>MIT License for testutils package</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

testutils/LICENCE.txt

• MIT License for the testutils package<br> • Copyright 2025 Apptly <br>Software Ltd with standard MIT terms

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-a4f99e6d9418bcf4b92eb882a658124f82ea4acae23f65ed857ab1ea9585c2be">+19/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Dependencies</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>go.sum</strong><dd><code>Add Go module dependency checksums for testutils package</code>&nbsp; </dd></summary>
<hr>

testutils/go.sum

• Adds Go module checksums for <code>darvaza.org/core</code>, <code>golang.org/x/net</code>, and <br><code>golang.org/x/text</code> dependencies

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-d2841921c1a3b8724e485d6ffa0a644f16d2ddf97b6adf92d7bc65adad1f8ce1">+6/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>go.mod</strong><dd><code>Create Go module definition for testutils package</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

testutils/go.mod

• Creates Go module definition for <code>darvaza.org/x/testutils</code> with Go <br>1.23.0 requirement<br> • Declares dependency on <code>darvaza.org/core v0.18.3</code> <br>with indirect dependencies

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-c41ce1d780ee13bbea0d4e9fafa21859bdbb8272e223ea9f911b7c5a1fd47d0b">+10/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>Makefile</strong><dd><code>Export TMPDIR environment variable in Makefile</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

Makefile

• Exports `TMPDIR` environment variable for build processes

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>cspell.json</strong><dd><code>Add testutils to spell checker dictionary</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal/build/cspell.json

• Adds "testutils" to the spell checker dictionary

</details>

  </td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-3094120fcb2e87cf154fce1053bc486a13eaa9d4200a24f858bc9c39ec2e0936">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Additional files</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>TESTING.md</strong></td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-7faab1222e23f72294d2a17d6985cb06e89fb8bd01cf9525fa2dd57157f35a56">+1931/-0</a></td>

</tr>

<tr>
  <td><strong>generic_gen.go</strong></td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/258/files#diff-1f1f41adafb0fb1e29ff7c65c36a9d4c7c2d1dacb7e8f5ccb1ea41fa83e13f76">+3513/-0</a></td>

</tr>
</table></details></td></tr></tr></tbody></table>

</details>

___

## Summary by CodeRabbit

- New Features
  - Added TestUtils: a public testing utilities package with generic test-case types, runners, helpers, and multi-pattern support for method/factory/function testing.

- Documentation
  - Comprehensive TestUtils docs, examples, license, README, architecture and release diagrams, and release instructions updated to include TestUtils.

- Tests
  - Large, multi-arity test suites and example tests exercising TestUtils across many scenarios to increase coverage.

- Chores
  - Exported TMPDIR to sub-make processes; added "testutils" to spell-check dictionary.

### CODE REVIEW

File: testutils/README.md [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+## Development
+
+For development guidelines, architecture notes, and AI agent instructions,
+see [AGENTS.md](AGENTS.md).
+
+## See Also
+
+* [TESTING.md](TESTING.md) for testing pattern guidelines.
+* [AGENTS.md](AGENTS.md) for development notes and architecture.
+* [Root TESTING.md](../../core/TESTING.md) for core testing requirements.
```
Comment (amery) [inline testutils/README.md:476]: invalid link

File: testutils/utils.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+// Make creates test cases based on the configuration.
+func (tc ArrayTestCaseData) Make() []core.TestCase {
+	testCases := make([]*DummyTestCase, tc.totalCount)
+
+	// Mark all as passing by default
+	for i := range tc.totalCount {
+		testName := fmt.Sprintf("test%d", i)
+		testCases[i] = NewDummyTestCase(testName, true)
+	}
```
[the remark this answers is not in the document]
Comment (amery) [reply testutils/utils.go:137]: Go 1.23 minimum version for this repository
